### PR TITLE
Add volume unit test

### PR DESCRIPTION
**Pull request by outscale-vbr**
* Add volume unit test foreach volume function

### CODE REVIEW

File: controllers/oscmachine_volume_controller_unit_test.go [hunk 1]
```diff
@@ -0,0 +1,784 @@
...
+			clusterSpec: defaultClusterInitialize,
+			machineSpec: infrastructurev1beta1.OscMachineSpec{
+				Node: infrastructurev1beta1.OscNode{
+					Volumes: []*infrastructurev1beta1.OscVolume{
+						{
+							Name:          "test-volume",
+							Iops:          1000,
+							Size:          50,
+							VolumeType:    "gp3",
+							SubregionName: "eu-west-2a",
```
Comment (outscale-mdr) [inline controllers/oscmachine_volume_controller_unit_test.go:374]: Also check standard and gp2

File: controllers/oscmachine_volume_controller_unit_test.go [hunk 2]
```diff
@@ -0,0 +1,784 @@
...
+		{
+			name:                        "failed to validate volume",
+			clusterSpec:                 defaultClusterReconcile,
+			machineSpec:                 defaultVolumeReconcile,
+			expVolumeFound:              true,
+			expValidateVolumeIdsErr:     fmt.Errorf("ValidateVolumeIds generic errors"),
+			expReconcileDeleteVolumeErr: fmt.Errorf("ValidateVolumeIds generic errors"),
+		},
+		{
+			name:                        "remove finalizer (user delete volume withut cluster-api)",
```
Comment (outscale-mdr) [inline controllers/oscmachine_volume_controller_unit_test.go:694]: Typo: "without"

File: controllers/oscmachine_volume_controller_unit_test.go [hunk 3]
```diff
@@ -0,0 +1,784 @@
...
+		{
+			name:                    "check volume exist (second tim reconcile loop)",
+			clusterSpec:             defaultClusterReconcile,
+			machineSpec:             defaultVolumeReconcile,
+			expVolumeFound:          true,
+			expValidateVolumeIdsErr: nil,
+			expReconcileVolumeErr:   nil,
+		},
+		{
+			name:                    "check two volumes exist (second timme reconcile loop)",
```
Comment (outscale-mdr) [inline controllers/oscmachine_volume_controller_unit_test.go:534]: Typo "time"

File: controllers/oscmachine_volume_controller_unit_test.go [hunk 4]
```diff
@@ -0,0 +1,784 @@
...
+	volumeTestCases := []struct {
+		name                    string
+		clusterSpec             infrastructurev1beta1.OscClusterSpec
+		machineSpec             infrastructurev1beta1.OscMachineSpec
+		expVolumeFound          bool
+		expValidateVolumeIdsErr error
+		expReconcileVolumeErr   error
+	}{
+		{
+			name:                    "check volume exist (second tim reconcile loop)",
```
Comment (outscale-mdr) [inline controllers/oscmachine_volume_controller_unit_test.go:526]: Typo "time"

**Review by outscale-mdr — CHANGES_REQUESTED** [review]

File: controllers/oscmachine_volume_controller_unit_test.go (hunk 1 above)
Comment (outscale-vbr) [reply controllers/oscmachine_volume_controller_unit_test.go:374]: Check Bad volumeType
I use volume that does not exist.

Comment (outscale-vbr) [reply controllers/oscmachine_volume_controller_unit_test.go:374]: Yes i miss it

**Review by outscale-mdr — APPROVED** [review]

**Comment by outscale-vbr** [comment]
close https://github.com/outscale-dev/cluster-api-provider-outscale/issues/28

**Pull request merged**